src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+from typing_extensions import override
+
+from openforms.contrib.objects_api.rendering import render_to_json
+
+# TODO: should be moved somewhere more generic?:
+from openforms.contrib.zgw.clients.utils import get_today
+from openforms.registrations.constants import REGISTRATION_ATTRIBUTE
+from openforms.submissions.mapping import MappingConfig, apply_data_mapping
+from openforms.submissions.models import Submission
+from openforms.variables.utils import get_variables_for_context
+
+from .typing import RegistrationOptions, RegistrationOptionsV1
+
+
+class BaseObjectsAPIRegistrator(ABC):
```

Comment:
The class hierarchy is a bit useless, but I didn't want to have 4 functions for both v1 and v2 & `get_record_data` and `get_update_payment_status_data`.

Considering no `__init__` method is defined, I might use a protocol instead.